src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const resArr = [];
+  let copyState = { ...state };
+
+  for (const instr of actions) {
+    const { type, extraData, keysToRemove } = instr;
+
+    switch (type) {
+      case 'addProperties':
+        for (const [key, value] of Object.entries(extraData)) {
+          copyState[key] = value;
+        }
+        resArr.push({ ...copyState });
+        break;
+      case 'removeProperties':
+        for (const key of Object.values(keysToRemove)) {
+          if (key in copyState) {
+            delete copyState[key];
+          }
+        }
+        resArr.push({ ...copyState });
+        break;
+      case 'clear':
+        copyState = {};
+        resArr.push({ ...copyState });
+        break;
```

Comment:
add `default` case (might be even empty - with `break` only) - that's a good practice

Reply:
hi, Even without copyState = { }
Just 
default:
break;
?

Reply:
@TarasHoliuk 
I understood simply 
```
default:
break;
```
, but what is the use of this ?, just wondering

Reply:
@LeoSpartaDeV  )
![Screenshot 2023-06-11 at 15 35 37](https://github.com/mate-academy/js_stateful-clones/assets/64746943/25f52396-b8ab-4cfb-a276-b9e496201ddd)

Reply:
Understood, thanks for example and material 👍🙌